lua/orgmode/org/mappings.lua

```diff
@@ -462,6 +462,147 @@ function OrgMappings:show_help()
   return Help.show()
 end
 
+function OrgMappings:edit_src_block()
+  local start_bufnr = vim.api.nvim_get_current_buf()
+  local file = Files.get_current_file()
+  local pos = vim.api.nvim_win_get_cursor(0)
+
+  local get_blocks = function()
```

Comment:
I don't think there's need to query all blocks in a document to find the right one. You can just get the current node via `file:get_node_at_cursor()` and traverse up the parent until you find the src block. Also consider that someone might do this on the exact SRC node so there's no need to traverse in that case.

Reply:
Updated, not super familiar with _Treesitter_ (and the functions that are available for each node) but I have changed to to function as you've mentioned here.